src/consensus/aft/raft.h

```diff
@@ -1461,7 +1669,7 @@ namespace aft
 
       for (auto node_id : to_remove)
       {
-        if (replica_state == Leader)
```

Comment:
We probably ought to move to a more detailed state, which would have a role (Leader/Follower/...) but also properties (executes, signs, ...) which can be set independently. The protocol variant would define which role has which proprties.